include/openthread/platform/trel.h

```diff
@@ -197,6 +198,9 @@ void otPlatTrelSend(otInstance       *aInstance,
  */
 extern void otPlatTrelHandleReceived(otInstance *aInstance, uint8_t *aBuffer, uint16_t aLength);
 
+
+const otTrelCounters *otPlatTrelGetCounters(void);
```

Comment:
We should clarify here, are the counters tracked by platform or OT core code?

Make sense to me that one entity does this. My preference is the OT core would track counters based on what it passes to platform later (which matches the current definitions of counters). 

If we want platform to track telemetry/counter info, we may need to define platform specific counters for this.

Reply:
Sorry, the previous commit was a draft, I pushed it just to test the changes through the build workflow. Now I added the clarification here.

Basically, the trel_interface cannot capture the trel packet transmission failures because otPlatTrelSend doesn't return error (the packet may only get queued and sent later), my previous commit wrongly tracked the error - the error there is not the packet transmisson error.